lib/membrane_rtc_engine/endpoints/hls/video_layout_maker.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Membrane.RTC.Engine.Endpoint.HLS.VideoLayoutMaker do
+  @moduledoc """
+  Module representing function for updating video layout for the HLS stream.
+  """
+  require Membrane.Pad
+
+  alias Membrane.Pad
+  alias Membrane.RTC.Engine.Track
+  alias Membrane.VideoCompositor.RustStructs.VideoPlacement
+
+  @spec update_layout(%{:tracks => map(), optional(any()) => any()}, Track.t()) :: [
+          {:forward, {:compositor, {:update_placement, list({Pad.ref_t(), VideoPlacement.t()})}}}
+        ]
+  def update_layout(state, curr_track) do
```

Comment:
A couple of thoughts here:
* we shouldn't pass the whole HLS endpoint state 
* we shouldn't return membrane action, it exposes internal HLS endpoint structure (e.g. `:compositor` name) which results in high modules coupling

Instead, I would propose following API:
```elixir
defmodule VideoLayoutMaker do
  @type state() :: any()
  @callback init() :: state()
  @callback track_added(state, Track.t()) :: {[{Track.id(), VideoPlacement.t()}], state}
  @callback track_removed(state, Track.t()) :: {[{Tack.id(), VideoPlacement.t()}], state}
end
```